### ISSUE TITLE
Ordinary dict addition produces -- Argument of type "Literal[' xx']" cannot be assigned to parameter 

### ISSUE THREAD

**Issue by mohsenBanan** [microsoft/pyright#2312]
The following:
-----------------
#!/bin/env python

icmInfo = { 'moduleDescription': [""" Some Text """], }

icmInfo['moduleUsage'] = """ Some Other Text """
-----------------

Results in:
-------------
mini.py     5   1 error    reportGeneralTypeIssues Argument of type "Literal[' Some Other Text ']" cannot be assigned to parameter "v" of type "list[str]" in function "__setitem__"
   "Literal[' Some Other Text ']" is incompatible with "list[str]" (lsp)
-------------

I am using pyright in emacs. 

pyright version is 1.1.163

This should be very simple to reproduce.

**Comment by erictraut** [comment]
I don't think you provided enough context. If I copy the line from your example, no errors are produced. Could you provide a bit more context — preferably a minimal self-contained sample that exhibits the error?

**Comment by mohsenBanan** [comment]
I have a file containing 5 lines:
---
#!/bin/env python

icmInfo = { 'moduleDescription': [""" Some Text """], }

icmInfo['moduleUsage'] = """ Some Other Text """

---
And that is the error that  I am getting.

**Comment by erictraut** [comment]
Ah, I misread your post because it wasn't formatted correctly. You can use triple ticks to surround code examples, like this:

```python
icmInfo = { 'moduleDescription': [""" Some Text """], }
icmInfo['moduleUsage'] = """ Some Other Text """
```

The behavior you're seeing is intended and is due to the type inference rules for dictionaries. Click [here](https://github.com/microsoft/pyright/blob/main/docs/type-inference.md#dictionary-expressions) for full documentation.

If your intent is to allow `icmInfo` to contain values of any type, you can provide a type annotation as follows:
```python
icmInfo: Dict[str, Any] = { 'moduleDescription': [""" Some Text """], }
icmInfo['moduleUsage'] = """ Some Other Text """
```

**Comment by mohsenBanan** [comment]
Okay. Strict static type checking! 
I am all good now.
Thanks Eric.
